### ISSUE TITLE
Ensure old lookup queues are cleaned up before creating new lookup queues

### ISSUE THREAD

**Issue by kbara** [Igalia/snabbswitch#691]
Make sure that no packets / resources get leaked under any circumstances.

Related to https://github.com/Igalia/snabb/pull/675 , which in turn is justified by https://github.com/Igalia/snabb/issues/655#issuecomment-265792331 .

**Comment by teknico** [comment]
TL;DR: nothing to see here, move along (hopefully). :-)

The change applied in #675 recreates the two lookup queues associated with the binding table when a reload message is received, inside the `LwAftr:push` method.

`LwAftr:push` unconditionally calls `flush_decapsulation`, `flush_encapsulation` and `flush_hairpin` each time it runs. These three functions in turn call `BTLookupQueue:process_queue` on both lookup queues, and then iterate over the queue elements calling `get_lookup` on each of them, thus emptying the queues.

The only other significant attribute of `BTLookupQueue` is a `LookupStreamer` instance, which in turn does not seem to contain, or point to, other resources.

To be on the safe side, in case `LwAftr:push` would be interrupted before flushing the queues, we could call the flush functions on the old lookup queues again, right before creating the new ones. Unfortunately, while the hairpin queue usage is unambiguous, `inet_lookup_queue` is used for both encapsulation and decapsulation: if the queue still contains any packets after exiting `LwAftr:push`, we don't know anymore whether we should encapsulate or decapsulate them.

If we exclude the possibility of `LwAftr:push` being interrupted before completion, I don't see a possibility for resource leaks here.

**Comment by kbara** [comment]
Having re-read the code, I accept your analysis. Thanks!
